packages/core/src/signals/Vector2CompoundSignalContext.ts

```diff
@@ -0,0 +1,247 @@
+import {InterpolationFunction, TimingFunction} from '../tweening';
+import {PossibleVector2, Vector2} from '../types';
+import {CompoundSignal, CompoundSignalContext} from './CompoundSignalContext';
+import {Signal} from './SignalContext';
+import {SignalExtensions, SignalGenerator, SignalValue} from './types';
+
+export type Vector2Signal<
+  TOwner = void,
+  TContext = Vector2CompoundSignalContext<TOwner>,
+> = CompoundSignal<PossibleVector2, Vector2, 'x' | 'y', TOwner, TContext> & {
+  edit(
+    callback: (current: Vector2) => SignalValue<PossibleVector2>,
+    duration?: number,
+    timingFunction?: TimingFunction,
+    interpolationFunction?: InterpolationFunction<Vector2>,
+  ): Vector2 | TOwner | SignalGenerator<PossibleVector2, Vector2>;
+  scale(
+    value: number,
+    duration?: number,
+    timingFunction?: TimingFunction,
+    interpolationFunction?: InterpolationFunction<Vector2>,
+  ): Vector2 | TOwner | SignalGenerator<PossibleVector2, Vector2>;
+  mul(
+    possibleVector: PossibleVector2,
+    duration?: number,
+    timingFunction?: TimingFunction,
+    interpolationFunction?: InterpolationFunction<Vector2>,
+  ): Vector2 | TOwner | SignalGenerator<PossibleVector2, Vector2>;
+  div(
+    possibleVector: PossibleVector2,
+    duration?: number,
+    timingFunction?: TimingFunction,
+    interpolationFunction?: InterpolationFunction<Vector2>,
+  ): Vector2 | TOwner | SignalGenerator<PossibleVector2, Vector2>;
+  add(
+    possibleVector: PossibleVector2,
+    duration?: number,
+    timingFunction?: TimingFunction,
+    interpolationFunction?: InterpolationFunction<Vector2>,
+  ): Vector2 | TOwner | SignalGenerator<PossibleVector2, Vector2>;
+  sub(
+    possibleVector: PossibleVector2,
+    duration?: number,
+    timingFunction?: TimingFunction,
+    interpolationFunction?: InterpolationFunction<Vector2>,
+  ): Vector2 | TOwner | SignalGenerator<PossibleVector2, Vector2>;
+  dot(
+    possibleVector: PossibleVector2,
+    duration?: number,
+    timingFunction?: TimingFunction,
+    interpolationFunction?: InterpolationFunction<Vector2>,
+  ): Vector2 | TOwner | SignalGenerator<PossibleVector2, Vector2>;
+  cross(
+    possibleVector: PossibleVector2,
+    duration?: number,
+    timingFunction?: TimingFunction,
+    interpolationFunction?: InterpolationFunction<Vector2>,
+  ): Vector2 | TOwner | SignalGenerator<PossibleVector2, Vector2>;
+  mod(
+    possibleVector: PossibleVector2,
+    duration?: number,
+    timingFunction?: TimingFunction,
+    interpolationFunction?: InterpolationFunction<Vector2>,
+  ): Vector2 | TOwner | SignalGenerator<PossibleVector2, Vector2>;
+};
+
+export class Vector2CompoundSignalContext<
```

Comment:
The class should be called just `Vector2SignalContext`
```suggestion
export class Vector2SignalContext<
```